src/passes/OptimizeInstructions.cpp

```diff
@@ -64,16 +64,57 @@ Index getMaxBits(Expression* curr, LocalInfoProvider* localInfoProvider) {
   } else if (auto* binary = curr->dynCast<Binary>()) {
     switch (binary->op) {
       // 32-bit
+      case RotLInt32:
+      case RotRInt32:
       case AddInt32:
       case SubInt32:
       case MulInt32:
-      case DivSInt32:
-      case DivUInt32:
-      case RemSInt32:
-      case RemUInt32:
-      case RotLInt32:
-      case RotRInt32:
+      case RemSInt32: {
+        if (auto* const_ = binary->right->dynCast<Const>()) {
+          auto maxBits = getMaxBits(binary->left, localInfoProvider);
+          if (maxBits == 32) {
+            return 32;
+          }
+          auto value = const_->value.geti32();
+          return value <= 1
+                   ? Index(0)
```

Comment:
this looks wrong. `x + 1` would reach this line and return `0` for any `x`, wouldn't it?

Reply:
It seems yes. Definitely need tests. For now I just remove this special path